Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -10,10 +11,26 @@ protected Piece(Player player)
         {
             Player = player;
             HasMoved = false;
+            CardinalDirections = new List<Tuple<int, int>>
+            {
+                new Tuple<int, int>(0, 1),
+                new Tuple<int, int>(0, -1),
+                new Tuple<int, int>(1, 0),
+                new Tuple<int, int>(-1, 0)
+            };
+            DiagonalDirections = new List<Tuple<int, int>>
+            {
+                new Tuple<int, int>(1, 1),
+                new Tuple<int, int>(1, -1),
+                new Tuple<int, int>(-1, -1),
+                new Tuple<int, int>(-1, 1)
+            };
         }
 
         public Player Player { get; private set; }
         public bool HasMoved { get; private set; }
+        public List<Tuple<int, int>> CardinalDirections { get; private set; }
+        public List<Tuple<int, int>> DiagonalDirections { get; private set; }
```

Comment:
Sorry, didn't spot this before. These variables should be declared abovec the constructor 🙂